application-xpoll-api/src/test/java/com/xwiki/xpoll/internal/DefaultXPollManagerTest.java

```diff
@@ -0,0 +1,121 @@
+import org.xwiki.test.junit5.mockito.MockComponent;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.web.XWikiRequest;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for {@link DefaultXPollManager}.
+ *
+ * @version $Id$
+ * @since 2.1
+ */
+@ComponentTest
+public class DefaultXPollManagerTest
+{
+    private static final String XPOLL_SPACE_NAME = "XPoll";
+
+    private static final LocalDocumentReference XPOLL_CLASS_REFERENCE =
+        new LocalDocumentReference(XPOLL_SPACE_NAME, "XPollClass");
+
+    private static final LocalDocumentReference XPOLL_VOTES_CLASS_REFERENCE =
+        new LocalDocumentReference(XPOLL_SPACE_NAME, "XPollVoteClass");
+
+    @InjectMockComponents
+    private DefaultXPollManager manager;
+
+    @MockComponent
+    private Provider<XWikiContext> contextProvider;
+
+    private XWikiContext xWikiContext;
+
+    private XWikiRequest request;
+
+    private XWiki wiki;
+
+    private XWikiDocument document;
+
+    private BaseObject xpollObj;
+
+    @BeforeEach
+    void setup()
+    {
+        this.xWikiContext = mock(XWikiContext.class);
+        when(this.contextProvider.get()).thenReturn(this.xWikiContext);
+    }
+
+    @Test
+    void urlMultipleSpaces()
+    {
+        this.request = mock(XWikiRequest.class);
+        when(this.xWikiContext.getRequest()).thenReturn(this.request);
+        when(this.request.getContextPath()).thenReturn("xwiki");
+
+        DocumentReference docRef = new DocumentReference("XWiki", Arrays.asList("Space1", "Space2"), "Page");
+
+        String result = manager.getRestURL(docRef);
+        String expected = "xwiki/rest/wikis/XWiki/spaces/Space1/spaces/Space2/pages/Page/xpoll";
+        assertEquals(expected, result);
+    }
+
+    @Test
+    void getVoteResults() throws XWikiException
+    {
+        this.xpollObj = mock(BaseObject.class);
+        this.document = mock(XWikiDocument.class);
+        this.wiki = mock(XWiki.class);
+
+        DocumentReference docRef = new DocumentReference("XWiki", Arrays.asList("Space1", "Space2"), "Page");
+
+        when(this.xWikiContext.getWiki()).thenReturn(this.wiki);
+        when(this.wiki.getDocument(docRef, this.xWikiContext)).thenReturn(this.document);
+        when(this.document.getXObject(XPOLL_CLASS_REFERENCE)).thenReturn(this.xpollObj);
+        when(this.xpollObj.getListValue("proposals")).thenReturn(Arrays.asList("Proposal1", "Proposal2", "Proposal3"));
+        when(this.document.getXObjects(XPOLL_VOTES_CLASS_REFERENCE)).thenReturn(new ArrayList<>());
+
+        Map<String, Integer> results = manager.getVoteResults(docRef);
+
+        assertEquals(3, results.size());
+        assertEquals(0, results.values().stream().reduce(0, Integer::sum));
```

Comment:
If the scores are ``[-1, 1, 0]`` or ``[-1, 2, -1]`` then this assert will pass :) I'd rather assert the map value set.